pkg/build/frontend.go

```diff
@@ -145,6 +146,22 @@ func resolveStates(ctx context.Context, bopts *BOpts, platform ocispecs.Platform
 				return
 			}
 
+			// if the platform is specified in the stage, use it
+			stagePlatform := platform
+			if stage.Platform != "" {
+				resolvedStagePlatform, err := shlex.ProcessWordWithMatches(stage.Platform, resolvedGlobalArgs)
+				if err != nil {
+					errCh <- fmt.Errorf("invalid platform for stage[%s]: %v", stage.BaseName, err)
+					return
+				}
+				platform, err := platforms.Parse(resolvedStagePlatform.Result)
```

Comment:
Use a different name for this variable. There’s an input arg named platform in this function already and this variable shadows it.

Reply:
Done in 6bf5db11a9aae432be315e43abbcc705cb6a6759